test/get_test.go

```diff
@@ -13,36 +14,158 @@ func TestGet(t *testing.T) {
 	ctx := context.Background()
 	client := newKine(t)
 
-	t.Run("FailMissing", func(t *testing.T) {
+	t.Run("FailNotFound", func(t *testing.T) {
 		g := NewWithT(t)
-		resp, err := client.Get(ctx, "testKey", clientv3.WithRange(""))
+		key := "testKeyFailNotFound"
 
+		// Get non-existent key
+		resp, err := client.Get(ctx, key, clientv3.WithRange(""))
+		g.Expect(err).To(BeNil())
+		g.Expect(resp.Kvs).To(BeEmpty())
+	})
+
+	t.Run("FailEmptyKey", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Get empty key
+		_, err := client.Get(ctx, "", clientv3.WithRange(""))
+		g.Expect(err).To(BeNil())
+	})
+
+	t.Run("FailRange", func(t *testing.T) {
+		g := NewWithT(t)
+		key := "testKeyFailRange"
+
+		// Get range with a non-existing key
+		resp, err := client.Get(ctx, key, clientv3.WithRange("thisIsNotAKey"))
 		g.Expect(err).To(BeNil())
 		g.Expect(resp.Kvs).To(BeEmpty())
 	})
 
 	t.Run("Success", func(t *testing.T) {
 		g := NewWithT(t)
+		key := "testKeySuccess"
 
 		// Create a key
 		{
 			resp, err := client.Txn(ctx).
-				If(clientv3.Compare(clientv3.ModRevision("testKey"), "=", 0)).
-				Then(clientv3.OpPut("testKey", "testValue")).
+				If(clientv3.Compare(clientv3.ModRevision(key), "=", 0)).
+				Then(clientv3.OpPut(key, "testValue")).
 				Commit()
 			g.Expect(err).To(BeNil())
 			g.Expect(resp.Succeeded).To(BeTrue())
 		}
 
 		// Get key
 		{
-			resp, err := client.Get(ctx, "testKey", clientv3.WithRange(""))
+			resp, err := client.Get(ctx, key, clientv3.WithRange(""))
 			g.Expect(err).To(BeNil())
 			g.Expect(resp.Kvs).To(HaveLen(1))
-			g.Expect(resp.Kvs[0].Key).To(Equal([]byte("testKey")))
+			g.Expect(resp.Kvs[0].Key).To(Equal([]byte(key)))
 			g.Expect(resp.Kvs[0].Value).To(Equal([]byte("testValue")))
 		}
 	})
+
+	t.Run("KeyRevision", func(t *testing.T) {
+		g := NewWithT(t)
+		key := "testKeyRevision"
+		var lastModRev int64
+
+		// Create a key with a known value
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision(key), "=", 0)).
+				Then(clientv3.OpPut(key, "testValue")).
+				Commit()
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+			lastModRev = resp.Responses[0].GetResponsePut().Header.Revision
+		}
+
+		// Get the key's version
+		{
+			resp, err := client.Get(ctx, key, clientv3.WithCountOnly())
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Count).To(Equal(int64(0)))
+		}
+
+		// Update the key
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision(key), "=", lastModRev)).
+				Then(clientv3.OpPut(key, "testValue2")).
+				Else(clientv3.OpGet(key, clientv3.WithRange(""))).
+				Commit()
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Header.Revision).To(Equal(lastModRev + 1))
```

Comment:
This is dangerous, could be any value greater than the old one, not necessarily incremented by 1. also below

See  https://github.com/canonical/kine/blob/38870b69012192f36718b8f528220680e68c27d5/test/update_test.go#L77